src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt

```diff
@@ -146,9 +141,18 @@ class NotificationService(
     */
     private fun sendNotification(quantumId: String, notificationGroup: List<ShiftNotification>) {
         val userPreference = userPreferenceService.getOrCreateUserPreference(quantumId)
+
+        data class Key(val shiftDate: LocalDateTime, val shiftType: String)
```

Comment:
This needs to be by QuantumId too, otherwise it's any shift on that day

Reply:
This is run on a `forEach` group of a collection that is grouped by `quantumId`. I thought I didn't need to group by them again as it should already be done. Not sure if I am thinking about this wrong though 🤷🏽‍♀️

Reply:
so long as a future developer knows that...do you think its a smell that you have to call it carefully?
belt and braces maybe? 😬 
maybe just a comment? because its in a private method you can call it however you want in the future.